src/conn/conn_dhandle.c

```diff
@@ -313,6 +313,15 @@ __wt_conn_dhandle_close(WT_SESSION_IMPL *session, bool final, bool mark_dead)
     btree = is_btree ? dhandle->handle : NULL;
 
     if (is_btree) {
+        /* Don't close the btree if there are changes not yet globally visible. */
+        /*
+         * !! Check if we have exclusive access here, else a transaction can start on this btree
+         * after this check !!
+         */
+        if (!WT_DHANDLE_IS_CHECKPOINT(dhandle) && (session != conn->sweep_session) &&
+          !__wt_txn_visible_all(session, btree->max_upd_txn, WT_TS_NONE))
+            return (EBUSY);
```

Comment:
Should this be

ret = EBUSY;
goto err;

if you want to handle the EBUSY error?

Reply:
It is done this way because this return is before anything is done that needs to be cleaned up.  If this branched to err unnecessary cleanup would be done.  For example, __wt_spin_lock(session, &dhandle->close_lock); is done between here and err: and unlocked by err. If we branch to err it would unlock a lock that is not locked.